gibberish_server/controllers/room_controller.js

```diff
@@ -74,16 +74,20 @@ class RoomController {
     static submitAnswer(req, res) {
         let roomId = req.body.roomId;
         let nickname = req.body.nickname;
-        let score = req.body.score;
+        let score = parseInt(req.body.score);
+        if (isNaN(score)) {
+            res.status(400).send('Score is not an integer')
+        }
         roomModel.getRoomInfoAsObject(roomId)
             .then(room => {
+                console.log('line80')
+                console.log(room.players)
                 let index = room.players.findIndex(player => player.name === nickname)
                 if (index !== -1) {
                     room.players[index].updateScore(score)
                 }
                 roomModel.saveRoom(room)
-                const { id, state, round, players, timer } = room
-                res.json({ id, state, round, players, timer })
+                res.json(room)
```

Comment:
is there a need to return the whole room? i picked the `{ id, state, round, players, timer}` because the room contains the qna data which is quite big and more importantly static, so to reduce the size of the GET response i only returned the variables that would change

Reply:
ohhh right, sure i'll change it